qiskit_nature/problems/sampling/protein_folding/qubit_op_builder.py

```diff
@@ -0,0 +1,544 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Builds qubit operators for all Hamiltonian terms in the protein folding problem."""
+from typing import Union
+
+import numpy as np
+from qiskit.opflow import OperatorBase, PauliOp, PauliSumOp
+
+from .bead_contacts.contact_map import ContactMap
+from .bead_distances.distance_map import DistanceMap
+from .exceptions.invalid_side_chain_exception import (
+    InvalidSideChainException,
+)
+from .exceptions.invalid_size_exception import InvalidSizeException
+from .penalty_parameters import PenaltyParameters
+from .peptide.pauli_ops_builder import _build_full_identity
+from .qubit_utils.qubit_fixing import _fix_qubits
+from .peptide.beads.base_bead import BaseBead
+from .peptide.peptide import Peptide
+
+
+# TODO link to a SamplingQubitOpBuilder interface
```

Comment:
Is this for the scope of this PR or for the future?

Reply:
Actually, I am not sure anymore if that interface is useful... Perhaps I would delete it for now.